src/assets/src/hooks/useEntityChanges.ts

```diff
@@ -0,0 +1,27 @@
+import { useState } from "react";
+
+import { ChangeEvent, compareEntities, ComparableEntity } from "../changes";
+
+
+export function useEntityChanges<T extends ComparableEntity>():
+    [ChangeEvent[], (oldEntities: readonly T[], newEntities: readonly T[]) => void, (key: number) => void]
+{
+    const [changeEvents, setChangeEvents] = useState([] as ChangeEvent[]);
+    const [nextID, setNextID] = useState(0);
+
+    const compareAndSetEventMap = (oldEntities: readonly T[], newEntities: readonly T[]): void => {
```

Comment:
Weird question, but I'm not sure what's meant by `Map` in the name of this function.

Reply:
Ah, nice catch, that is a vestige of a prior implementation that was using an object with IDs as the state variable instead of an array like it currently is. I need to update the function name, thanks!

Reply:
Should be resolved by https://github.com/tl-its-umich-edu/remote-office-hours-queue/pull/258/commits/2beee7b9579a0011cac4e7eb830a0e1be431ce76.